### PR TITLE
Added backward transform - Linear Layer

### PR DESCRIPTION
**Pull request by JeffreyWong20**
This PR implements a basic transformation to modify the backward pass of nodes in a computational graph, currently supporting only `Linear` modules. This might not be the best way to implement the transform pass. Please leave your views down!

> [!IMPORTANT]
> This PR wraps all `Linear` modules with their functional equivalents. As a result, quantizer information at the module level is now lost. Although this information theoretically isn't necessary, testing is required to confirm that this change does not break the PTQ pathway.

- [ ] Extend support to other modules (e.g., `Conv`, `ReLU`)
- [ ] Apply similar transformations for other functional components

### CODE REVIEW

**Review by ChengZhang-98 — COMMENTED** [review]
⛏️ Well done

File: test/passes/graph/transforms/training/test_training_base_template.py [hunk 1]
```diff
@@ -0,0 +1,221 @@
+from functools import partial
```
[the remark this answers is not in the document]
Comment (JeffreyWong20) [reply test/passes/graph/transforms/training/test_training_base_template.py]: Yes, this is a file to test forward and backward can indeed be explicitly modified and work as expected. Renamed to test_training_IntergerQuantize.

**Comment by Aaron-Zhao123** [comment]
@ChengZhang-98 if you finished reviewing it, maybe just merge this to main?

**Review by ChengZhang-98 — APPROVED** [review]

**Comment by ChengZhang-98** [comment]
@JeffreyWong20 You may merge it into main if you've finished this PR

**Pull request merged**